### DISCUSSION TITLE
q-table body-cell-name with password/text switch for each row

### DISCUSSION THREAD

**Discussion by TribesTom** [quasarframework/quasar#9671] [General - Components / Directives / etc] [answered]
Hello,

I'm trying to make q-table with a body-cell-name with a q-input which is in password type by default and that I can switch to text mode with an icon.

To do this, I add a property on each of my array object with password or text when retreiving from backend.
And when I click the icon, it change this property.

I have test a lot of things but without success.

Here is the code : 

`      <q-table :data="customers_list" :columns="customers_columns" :columns_filter="true" :loading="search_loading" :pagination="customer_pagination" >
        <template v-slot:body-cell-tacacs_key="props">
            <q-td :props="props">
                <q-input v-model="props.value" :type="props.row.tacacs_key_visiblity" borderless readonly :props="props">
                  <template v-slot:append>
                    <q-icon
                      :name="props.row.tacacs_key_visiblity == 'text' ? 'visibility' : 'visibility_off'"
                      class="cursor-pointer"
                      @click="toggleTacacsVisibility(props)"
                    />
                  </template>
                </q-input>
            </q-td>
        </template>
        <template v-slot:loading>
            <q-inner-loading showing color="primary" />
        </template>
      </q-table>`

scripts : 

`   toggleTacacsVisibility(props){
      console.log(this.customers_list.indexOf(props.row))

      if (props.row.tacacs_key_visiblity == "password") {
        console.log("1")
        props.row.tacacs_key_visiblity = "text"
        this.customers_list[this.customers_list.indexOf(props.row)].tacacs_key_visiblity = "text"
      } else {
        props.row.tacacs_key_visiblity = "password"
        console.log("2")
        this.customers_list[this.customers_list.indexOf(props.row)].tacacs_key_visiblity = "password"
      }
      console.log(props)
    }`

I can see in the console that the props is updating but on frontend I don't see it.

How could I make this work ?

Thanks

**Comment by ontwikkelfabriek** [comment, marked as answer]
@TribesTom 

> I can see in the console that the props is updating but on frontend I don't see it.

Assuming you're using **Qv1** with **vue2** (**vue3** does **not** have this issue) 

See `Change Detection Caveats` for `object` and `arrays`
https://vuejs.org/v2/guide/reactivity.html

**Reply by TribesTom** [reply]
Hello,

Thanks you for the help.

I did find the way to do it.

For info if someone come here with a search : 

`    toggleTacacsVisibility(props){

      if (props.row.tacacs_key_visiblity == "password") {
        this.$set(props.row, 'tacacs_key_visiblity',"text")
      } else {
        this.$set(props.row, 'tacacs_key_visiblity',"password")
      }

      this.$set(this.customers_list,this.customers_list.indexOf(props.row),props.row)
    }`

I don't know if it's the best way but It run good.
